bindings/python/setup.py

```diff
@@ -72,18 +58,48 @@ def append_libbson_flags(module):
         module.extra_link_args += ["-Wl,-rpath,$ORIGIN"]
 
     # https://cython.readthedocs.io/en/latest/src/tutorial/external.html#dynamic-linking
-    lnames = query_pkgconfig("pkg-config --libs-only-l {}".format(pc_path)).split()
+    lnames = lnames.split()
     # Strip whitespace to avoid weird linker failures on manylinux images
     libnames = [lname.lstrip('-l').strip() for lname in lnames]
     module.libraries.extend(libnames)
 
 
 def append_arrow_flags(module):
+    import numpy as np
+    import pyarrow as pa
+
     module.include_dirs.append(np.get_include())
     module.include_dirs.append(pa.get_include())
-    module.libraries.extend(pa.get_libraries())
     module.library_dirs.extend(pa.get_library_dirs())
 
+    # Handle the arrow library files manually.
+    # Alternative to using pyarrow.create_library_symlinks().
+    # You can use MONGO_ARROW_LIBDIR to explicitly set the location of the 
+    # arrow libraries (for instance in a conda build).
+    # We first check for an unmodified path to the library,
+    # then look for a library file with a version modifier, e.g. libarrow.600.dylib.
+    arrow_lib = os.environ.get('MONGO_ARROW_LIBDIR', pa.get_library_dirs()[0])
+    if platform == "darwin":
+        patterns = ['.dylib', '.[0-9][0-9][0-9].dylib']
+    elif platform == 'linux':
+        patterns = ['.so', '.so.[0-9][0-9][0-9]']
```

Comment:
What happens when pyarrow 10 comes out? would this need to change to `'.so.[0-9][0-9][0-9][0-9]'`? Is there a way to future proof that?

Reply:
Updated to add regex handling

Reply:
Updated again to working glob syntax with wildcards